src/apphosting/githubConnections.ts

```diff
@@ -98,39 +99,55 @@ export async function linkGitHubRepository(
   utils.logBullet(clc.bold(`${clc.yellow("===")} Import a GitHub repository`));
   // Fetch the sentinel Oauth connection first which is needed to create further GitHub connections.
   const oauthConn = await getOrCreateOauthConnection(projectId, location);
-  const existingConns = await listAppHostingConnections(projectId, location);
+  let installationId = await promptGitHubInstallation(projectId, location, oauthConn);
 
-  if (existingConns.length === 0) {
-    existingConns.push(
-      await createFullyInstalledConnection(projectId, location, generateConnectionId(), oauthConn),
+  while (installationId === ADD_ACCOUNT_CHOICE) {
+    utils.logBullet(
+      "Install the Firebase App Hosting GitHub app on a new account to enable access to those repositories",
```

Comment:
nit* not blocking

I wonder how surprising it is for users to ultimately be redirected to DevConnect. It looks friendly enough, and even says you should close the page, but I wonder if we should give a hint here about what to expect when a popup is open.

Reply:
hmm we could add some messaging here but I feel that this would make it too verbose. Ideally we redirect to a firebase branded success page which we can look into separately.